tests/uts/test_agent.py

```diff
@@ -0,0 +1,92 @@
+import json
+import os
+import unittest
+import uuid
+from threading import Thread
+from time import sleep, time
+
+from fakeredis import FakeStrictRedis
+from mock import patch
+
+from talker_agent.talker import TalkerAgent
+
+WAIT_FOR_EXCEPTION_TIMEOUT = 1
+WAIT_FOR_EXCEPTION_ITERATION_SLEEP = 0.0001
+
+
+def get_uuid():
+    return str(uuid.uuid4())
+
+
+def kill_process_before_signaling(self, job_id, signal):
+    job = self.current_processes.get(job_id)
+    os.killpg(os.getpgid(job.popen.pid), signal)
+    while job.popen.poll() is None:
+        pass
+    job.send_signal(signal)
+
+
+def raise_file_not_found(*args, **kwargs):
+    raise OSError(2, 'No such file or directory')
+
+
+JOBS_DIR = '/tmp/talker/jobs'
+
+
+@patch('talker_agent.talker.JOBS_DIR', JOBS_DIR)
+class TestClient(unittest.TestCase):
+
+    def setUp(self):
+        super(TestClient, self).setUp()
+
+        if not os.path.isdir(JOBS_DIR):
+            os.makedirs(JOBS_DIR)
+
+        self.agent = TalkerAgent()
+        self.agent.host_id = get_uuid()
+        self.agent.redis = FakeStrictRedis()
+        self.agent.pending_exception = None
+        self.exception_class = None
+        self.agent_thread = Thread(target=self.run_agent)
+        self.agent_thread.start()
+
+    def tearDown(self):
+        self.agent.kill()
+        self.agent_thread.join()
+
+    def run_agent(self):
+        try:
+            self.agent.start()
+        except Exception as e:
+            self.exception_class = e.__class__
```

Comment:
lets be more specific. keep the exception itself and reraise it inside a `assertRaises`